### ISSUE TITLE
Request to Include workbench.action.terminal.killAll in terminal.integrated.commandsToSkipShell by Default

### ISSUE THREAD

**Issue by q-l-p** [microsoft/vscode#174308]
Type: <b>Feature Request</b>

Dear VSCode Team,

I respectfully request the inclusion of the `workbench.action.terminal.killAll` command in the `terminal.integrated.commandsToSkipShell` list by default. Currently, when trying to bind this command to a function key, the shortcut only works if the terminal is not focused. This leads to a frustrating experience for users who simply want to bind this command to a function key. This problem persists in Visual Studio Code - Insiders.

A workaround to this issue is to bind the command to a non-function key or to manually add it to the `terminal.integrated.commandsToSkipShell` list. But this process could be time-consuming. It took me days of trial and error, along with numerous internet searches and AI queries, to finally find a solution to this issue.

Including the `workbench.action.terminal.killAll` command in the `terminal.integrated.commandsToSkipShell` list by default will greatly streamline the workflow for users, allowing for more convenient and efficient use of VSCode. This change would enhance the overall user experience and give VSCode a more premium feel.

Thank you for considering this request. I believe it would make VSCode an even more powerful and user-friendly tool.

Best regards,

James Smith

VS Code version: Code 1.75.1 (441438abd1ac652551dbe4d408dfcec8a499b8bf, 2023-02-08T21:32:34.589Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

**Comment by meganrogge** [comment]
This works for me. Try running `Developer: Toggle Keyboard Shortcuts Troubleshooting` to see what's going on

**Comment by q-l-p** [comment]
When you say, "This works for me", what are you referring to? Are you saying that you can bind a function key to the `workbench.action.terminal.killAll` command and that the shortcut works even with the terminal focused?

I was about to submit a bug report, but after reading [this](https://github.com/microsoft/vscode/issues/7269#issuecomment-238286922) I thought function key shortcuts are not supposed to work when the terminal is focused.

I submitted this request because it's too much of a hassle to bind a function key to the `workbench.action.terminal.killAll` command. The shortcut won't work if the terminal is focused. The workaround is to manually add the command to the `terminal.integrated.commandsToSkipShell` list. The list already contains the command `workbench.action.terminal.kill` but not the `killAll` command. 

I just tried to prevent others from going through all these hoops in order to create a keyboard shortcut.

**Comment by meganrogge** [comment]
Yes without adding that to commands to skip shell, I can use a keyboard shortcut to kill all terminals with the terminal focused. It looks like the shortcut you have configured isn't working based on the logs. I'd try changing that

**Comment by q-l-p** [comment]
Just to make sure everything is clear. Are you able to bind the `F5` key (or any other function key) to the command `workbench.action.terminal.killAll` and kill all terminals while the terminal is focused by pressing `F5` without adding the command to `terminal.integrated.commandsToSkipShell`? Because, on my system, I'm unable to do this, such a shortcut would only work if the terminal is _not_ focused. Or, the shortcut works even with the terminal focused if a non-function key is bound to the command (like, for example `ctrl+'`). I thought this is by design, especially after reading [this](https://github.com/microsoft/vscode/issues/7269#issuecomment-238286922).

Now, I don't know what to do. I already found a workaround. I only posted here thinking about others who might run into this problem. I have Windows 10 (64-bit) and a fresh install of Visual Studio Code. And, for me, this issue also happens in the Insiders version. Should I post a bug report or just leave it alone?
